sway-core/src/parse_tree/declaration/variable.rs

```diff
@@ -10,3 +18,231 @@ pub struct VariableDeclaration {
     pub body: Expression, // will be codeblock variant
     pub is_mutable: bool,
 }
+
+pub enum VariableDeclarationLHS {
```

Comment:
could we provide some documentation for this? As with all fully-`pub` structs -- we currently have pretty good coverage in other areas